### PR TITLE
refactor checkpoint class sync

### PR DESCRIPTION
**Pull request by CHr15F0x**
Title :arrow_up: .

### CODE REVIEW

File: crates/storage/src/connection/block.rs [hunk 1]
```diff
@@ -476,6 +477,8 @@ impl Transaction<'_> {
             ret.push(cc);
         }
 
+        ret.reverse();
+
```
Comment (CHr15F0x) [inline crates/storage/src/connection/block.rs:481]:
```suggestion
```

File: crates/storage/src/connection/block.rs [hunk 2]
```diff
@@ -440,6 +440,7 @@ impl Transaction<'_> {
         Ok(ret)
     }
 
+    /// Items are sorted in descending order.
```
Comment (CHr15F0x) [inline crates/storage/src/connection/block.rs:443]:
```suggestion
```

File: crates/pathfinder/src/sync/checkpoint.rs [hunk 3]
```diff
@@ -322,31 +320,40 @@ async fn handle_state_diff_stream(
...
+            10,
+        )
+        .pipe(
+            class_definitions::CompileSierraToCasm::new(fgw, tokio::runtime::Handle::current()),
+            10,
+        )
+        .pipe(class_definitions::Store(storage.connection()?), 10)
+        .into_stream()
+        // FIXME
+        //.inspect_ok(|x| tracing::info!(tail=%x.data, "Class definitions chunk synced"))
```
Comment (kkovaacs) [inline crates/pathfinder/src/sync/checkpoint.rs:351]: Why is this FIXME here?

**Review by kkovaacs — COMMENTED** [review]
LGTM other than that FIXME.

**Review by Mirko-von-Leipzig — APPROVED** [review]

**Review by kkovaacs — APPROVED** [review]

**Pull request merged**